packages/trpc/server/document-router/router.ts

```diff
@@ -97,15 +99,18 @@ export const documentRouter = router({
       }
     }),
 
-  deleteDraftDocument: authenticatedProcedure
+  deleteDocument: authenticatedProcedure
     .input(ZDeleteDraftDocumentMutationSchema)
     .mutation(async ({ input, ctx }) => {
       try {
-        const { id } = input;
+        const { id, status } = input;
 
         const userId = ctx.user.id;
 
-        return await deleteDraftDocument({ id, userId });
+        if (status === DocumentStatus.DRAFT) {
+          return await deleteDraftDocument({ id, userId });
+        }
+        return await deleteNonDraftDocument({ id, userId, status });
```

Comment:
Should we simplify this by merging the two and just returning early for draft documents? We typically don't introduce logic in our tRPC routers as doing so makes them less portable for other API types.

Reply:
oh okay, I'll just make a `deleteDocument({ id, userId, status })` function that merges the functionality